chat.go

```diff
@@ -43,12 +53,70 @@ type ChatCompletionRequest struct {
 	FrequencyPenalty float32                 `json:"frequency_penalty,omitempty"`
 	LogitBias        map[string]int          `json:"logit_bias,omitempty"`
 	User             string                  `json:"user,omitempty"`
+	Functions        []*FunctionsDefine      `json:"functions,omitempty"`
+	FunctionCall     string                  `json:"function_call,omitempty"`
 }
 
+type FunctionsDefine struct {
+	Name        string `json:"name"`
+	Description string `json:"description,omitempty"`
+	// it's required in function call
+	Parameters *FuncParams `json:"parameters"`
+}
+
+type FuncParams struct {
```

Comment:
How about `FunctionsParam` to match the naming pattern of `FunctionsDefine`?
If `FunctionsDefine` is to be changed to singular `FunctionDefine`, then `FunctionParams` looks good.